cmd/juju/cloud/list_test.go

```diff
@@ -164,43 +173,35 @@ func (s *listSuite) assertListTabular(c *gc.C, expectedOutput string) {
 			},
 		},
 	}
-	var controllerAPICalled string
 	cmd := cloud.NewListCloudCommandForTest(
 		s.store,
-		func(controllerName string) (cloud.ListCloudsAPI, error) {
-			controllerAPICalled = controllerName
+		func() (cloud.ListCloudsAPI, error) {
 			return s.api, nil
 		})
 
 	ctx, err := cmdtesting.RunCommand(c, cmd, "--controller", "mycontroller", "--format", "tabular")
 	c.Assert(err, jc.ErrorIsNil)
 	s.api.CheckCallNames(c, "Clouds", "Close")
-	c.Assert(controllerAPICalled, gc.Equals, "mycontroller")
+	c.Assert(cmd.ControllerName, gc.Equals, "mycontroller")
 
-	out := cmdtesting.Stderr(ctx)
-	out = strings.Replace(out, "\n", "", -1)
-	c.Assert(out, gc.Matches, `Clouds on controller "mycontroller":.*`)
-
-	out = cmdtesting.Stdout(ctx)
-	c.Assert(out, jc.DeepEquals, expectedOutput)
+	out := cmdtesting.Stdout(ctx)
+	c.Assert(out, jc.Contains, expectedOutput)
 }
 
 func (s *listSuite) TestListTabular(c *gc.C) {
 	s.assertListTabular(c, `
-Cloud    Regions  Default  Type       Description
-antnest        1  default  openstack  
-beehive        1  default  k8s        
-
+Cloud           Regions  Default          Type        Description
```

Comment:
So save repeating the description, we can print "from controller foo" at the top of the table. There will only ever be one controller, especially if we do  not include local results unless asked for. If we do go with allowing both local and controller if specified by the user, we can print them in separate tables; the controller and local clouds do serve different purposes and it makes sense to look at them separately.